ui/src/shared/components/TimeMachineQueryBuilder.test.tsx

```diff
@@ -0,0 +1,97 @@
+import React from 'react'
+
+import {renderWithRedux} from 'src/mockState'
+import {source} from 'mocks/dummyData'
+import {waitForElement, fireEvent} from 'react-testing-library'
+
+import TimeMachineQueryBuilder from 'src/shared/components/TimeMachineQueryBuilder'
+
+jest.mock('src/shared/apis/v2/queryBuilder')
+
+let setInitialState
+describe('TimeMachineQueryBuilder', () => {
+  beforeEach(() => {
+    setInitialState = state => {
+      return {
+        ...state,
+        sources: {
+          activeSourceID: source.id,
+          sources: {
+            [source.id]: source,
+          },
+        },
+      }
+    }
+  })
+
+  it('can select a bucket', async () => {
+    const {getByTestId, getAllByTestId, queryAllByTestId} = renderWithRedux(
+      <TimeMachineQueryBuilder />,
+      setInitialState
+    )
+
+    let bucketsDropdownClosed = await waitForElement(() =>
+      getByTestId('buckets--button')
+    )
+
+    fireEvent.click(bucketsDropdownClosed)
+
+    const bucketItems = getAllByTestId(/dropdown--item/)
+
+    expect(bucketItems.length).toBe(2)
+
+    const b2 = getByTestId('dropdown--item b2')
```

Comment:
I wonder if we need to use the `getByTestId` helper so liberally. It seems to be described as a "last resort" sort of selector by react-testing-library if there's nothing else we can use.